lib/Segments/SegmentSubscribersRepository.php

```diff
@@ -189,9 +189,8 @@ private function filterSubscribersInDynamicSegment(
     // For BC compatibility fetching an empty result
     if (count($filters) === 0) {
       return $queryBuilder->andWhere('0 = 1');
-    }
-    foreach ($filters as $filter) {
-      $queryBuilder = $this->filterHandler->apply($queryBuilder, $filter);
+    } elseif ($segment instanceof SegmentEntity) {
```

Comment:
This change will cause that this method will not work correctly when $segmentEntity is not passed an it was expected to work. It is used here https://github.com/mailpoet/mailpoet/blob/7e0c718683e98d3c26fb14ef0995a6a23d868ba4/lib/Segments/SegmentSubscribersRepository.php#L80

And it results in broken functionality for showing subscriber count during creation of a dynamic segment.